tests/profile/test_user_configurable_profiler_modular.py

```diff
@@ -0,0 +1,592 @@
+    assert profiler.value_set_threshold == "MANY"
+    assert not profiler.table_expectations_only
+    assert profiler.excluded_expectations == []
+
+
+def test_profiler_init_full_config_no_semantic_types(cardinality_validator):
+    """
+    What does this test do and why?
+    Confirms that profiler initializes properly with a full config, without a semantic_types dict
+    """
+
+    profiler = UserConfigurableProfiler(
+        cardinality_validator,
+        primary_or_compound_key=["col_unique"],
+        ignored_columns=["col_one"],
+        value_set_threshold="UNIQUE",
+        table_expectations_only=False,
+        excluded_expectations=["expect_column_values_to_not_be_null"],
+    )
+    assert profiler.primary_or_compound_key == ["col_unique"]
+    assert profiler.ignored_columns == [
+        "col_one",
+    ]
+    assert profiler.value_set_threshold == "UNIQUE"
+    assert not profiler.table_expectations_only
+    assert profiler.excluded_expectations == ["expect_column_values_to_not_be_null"]
+
+    assert "col_one" not in profiler.column_info
+
+
+def test_init_with_semantic_types(cardinality_validator):
+    """
+    What does this test do and why?
+    Confirms that profiler initializes properly with a full config and a semantic_types dict
+    """
+
+    semantic_types = {
+        "numeric": ["col_few", "col_many", "col_very_many"],
+        "value_set": ["col_two", "col_very_few"],
+    }
+    profiler = UserConfigurableProfiler(
+        cardinality_validator,
+        semantic_types_dict=semantic_types,
+        primary_or_compound_key=["col_unique"],
+        ignored_columns=["col_one"],
+        value_set_threshold="unique",
+        table_expectations_only=False,
+        excluded_expectations=["expect_column_values_to_not_be_null"],
+    )
+
+    assert "col_one" not in profiler.column_info
+
+    # assert profiler.column_info.get("col_none") == {
+    #     "cardinality": "NONE",
+    #     "type": "NUMERIC",
+    #     "semantic_types": [],
+    # }
+    assert profiler.column_info.get("col_two") == {
+        "cardinality": "TWO",
+        "type": "INT",
+        "semantic_types": ["VALUE_SET"],
+    }
+    assert profiler.column_info.get("col_very_few") == {
+        "cardinality": "VERY_FEW",
+        "type": "INT",
+        "semantic_types": ["VALUE_SET"],
+    }
+    assert profiler.column_info.get("col_few") == {
+        "cardinality": "FEW",
+        "type": "INT",
+        "semantic_types": ["NUMERIC"],
+    }
+    assert profiler.column_info.get("col_many") == {
+        "cardinality": "MANY",
+        "type": "INT",
+        "semantic_types": ["NUMERIC"],
+    }
+    assert profiler.column_info.get("col_very_many") == {
+        "cardinality": "VERY_MANY",
+        "type": "INT",
+        "semantic_types": ["NUMERIC"],
+    }
+    assert profiler.column_info.get("col_unique") == {
+        "cardinality": "UNIQUE",
+        "type": "INT",
+        "semantic_types": [],
+    }
```

Comment:
Since the `self.all_table_columns` attribute is new and is important for the rest of class functionality, would be nice to also have an assertion for this value in these tests. 

Also, since the ability to pass in a `Batch` is also new, I think we should add one more fixture+test that tests passing in only the `Batch` (this would have caught `self.all_table_columns` not being set properly per my comment above, if I'm correct).